### ISSUE TITLE
ambiguous use of 'nan'

### ISSUE THREAD

**Issue by opticalgenesis** [PureSwift/Silica#6]
Trying to build a package with Cacao (dependent on Silica), I'm hit with a fatal compile error complaining about an ambiguous use of 'nan'.
![screenshot from 2017-10-12 14 01 13](https://user-images.githubusercontent.com/9084974/31497127-e6ce2bf4-af55-11e7-859a-1bea578501f8.png)

**Comment by vanlepthien** [comment]
Only encountered on Linux builds 
```swift
public extension CGRect{
\#if os(Linux)
...
    public static var null: CGRect { return CGRect(x: .nan, y: .nan, width: .nan, height: .nan) }
...
\#endif
}
```
If CGFloat is specified, Silica compiles successfully:
```swift
    public static var null: CGRect { return CGRect(x: CGFloat.nan, y: CGFloat.nan, width: CGFloat.nan, height: CGFloat.nan) }
```
The Apple (non-extended version) of CGRect has multiple init methods, allowing either Double and CGFloat values to be passed. The unqualified use of .nan makes the reference ambiguous.

**Comment by colemancda** [comment]
Fixed in [v1.3.0](https://github.com/PureSwift/Silica/releases/tag/1.3.0)
